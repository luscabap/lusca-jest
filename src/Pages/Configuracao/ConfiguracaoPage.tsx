import Cabecalho from "../../components/Cabecalho";
import Formulario from "../../components/Formulario/Formulario";
import Rodape from "../../components/Rodape/Rodape";
import styles from './ConfiguracaoPage.module.scss'

const ConfiguracaoPage = () => {
    return (
        <div className={styles.container}>
            <Cabecalho />
                <div className={styles.container__infos}>
                    <Formulario />
                    <Rodape />
                </div>
        </div>
    )
};

export default ConfiguracaoPage;