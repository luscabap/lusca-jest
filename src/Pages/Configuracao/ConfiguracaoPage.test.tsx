import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import ConfiguracaoPage from "./ConfiguracaoPage";

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
})

describe('a página de configuração', () => {

    test('deve ser renderizada corretamente', () => {
        const { container } = render (
            <RecoilRoot>
                <ConfiguracaoPage />
            </RecoilRoot>
        )

        expect(container).toMatchSnapshot();

    })
})