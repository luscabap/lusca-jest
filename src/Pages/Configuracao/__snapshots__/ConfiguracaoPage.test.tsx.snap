// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <img
        alt="Logo da página"
        src="logo.png"
      />
      <img
        alt="Foto da participante"
        class="container__participante"
        src="participante.png"
      />
    </div>
    <div
      class="container__infos"
    >
      <form
        class="container"
      >
        <h2
          class="container__titulo"
        >
          Vamos começar!
        </h2>
        <div
          class="container__containerBotoes"
        >
          <input
            class="container__containerBotoes__input"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="container__containerBotoes__botao"
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <footer
        class="container"
      >
        <button
          class="container__botao"
          disabled=""
        >
          Iniciar brincadeira!
        </button>
        <img
          alt="Imagem de 2 sacolas verde e azul"
          src="sacolas.png"
        />
      </footer>
    </div>
  </div>
</div>
`;
